agenthub/codeact_agent/codeact_agent.py

```diff
@@ -151,19 +147,23 @@ def step(self, state: State) -> Action:
                 assert isinstance(
                     prev_action, self.SUPPORTED_ACTIONS
                 ), f'{prev_action.__class__} is not supported (supported: {self.SUPPORTED_ACTIONS})'
-                # prev_action is already added to self.messages when returned
+                if (
+                    isinstance(prev_action, MessageAction)
+                    and prev_action.source == 'user'
+                ):
+                    self.messages.append(
+                        {'role': 'user', 'content': prev_action.content}
+                    )
+                    if obs.content.strip() == '/exit':
```

Comment:
```suggestion
                    if prev_action.content.strip() == '/exit':
```

Reply:
This one is necessary, obs is null obs now.

Reply:
Whoops, another good catch!